.github/workflows/main_win.yaml

```diff
@@ -187,6 +187,43 @@ jobs:
           type: onhost
           os: windows
 
+  # Runs on-device integration and unit tests.
+  on-device-test:
+    needs: [initialize, build]
+    # Run ODT when on_device label is applied on PR.
+    # Also, run ODT on push and schedule if not explicitly disabled via repo vars.
+    if: |
+      fromJSON(needs.initialize.outputs.on_device_test).enabled == true && ((
+        github.event_name == 'pull_request' &&
+        contains(github.event.pull_request.labels.*.name, 'on_device') ) || ((
+            inputs.nightly == 'true' || github.event_name == 'schedule') &&
+            vars.RUN_ODT_TESTS_ON_NIGHTLY != 'False') ||
+          ( github.event_name == 'push' && vars.RUN_ODT_TESTS_ON_POSTSUBMIT != 'False' ) )
+    runs-on: [self-hosted, odt-runner]
+    name: ${{ matrix.name }}_on_device_${{ matrix.shard }}
+    container: ${{ needs.docker-unittest-image.outputs.docker_unittest_tag }}
```

Comment:
There's another section of code missing from the main workflow. The unit test image isn't being built in the windows pipeline. Once you add the shards it will fail here.

For now just copy the step from there.